py-m2crypto: update to 0.39.0

aque opened the pull request:
#### Description
Closes: https://trac.macports.org/ticket/66337

<!-- Note: it is best to make pull requests from a branch rather than from master -->

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion) $(uname -m)"; xcode=$(xcodebuild -version 2>/dev/null); if [ $? == 0 ]; then echo "$(echo "$xcode" | awk '\''NR==1{x=$0}END{print x" "$NF}'\'')"; else echo "Command Line Tools $(pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '\''/version:/ {print $2}'\'')"; fi' | tee /dev/tty | pbcopy
-->
macOS 11.7.10 20G1427 x86_64
Xcode 13.2.1 13C100

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

reneeotten commented:
@aque please check/fix the CI failure(s).

aque commented:
We [ran into this issue](https://trac.macports.org/ticket/66337) with version 0.38. I Googled for `"OPENSSL_sk_find_all" "m2crypto"` and the only result was that ticket. This error is somehow limited to py27 and py37, but I could not find anything version-specific in that part of the code. [Upstream has also deprecated py27](https://gitlab.com/m2crypto/m2crypto/-/blob/master/CHANGES) and plans to remove it in their next release. Based on all this, I removed `py27` and `py37` from `${python.versions}` and hope that is an acceptable solution.

aque commented:
Apologies for the multiple pushes. I wondered if parallel jobs might be the issue and forgot to reset my branch.

reneeotten commented:
removing the py27/py37 subports is only acceptable if they have no dependents. You check that with "port depends:pyXY-<portname>

aque commented:
Thanks @reneeotten. There were none for py37 but 3 for py27. What is the path forward in this case? Can I find a working version for py27 and override `${version}` and `${checksums}` via an `if` statement?

```
$ port echo depends:py27-m2crypto
dqsegdb
ligo-lvalert
py27-euca2ools
```

reneeotten commented:
upstream claims it should work with all Python versions based on their setup.py classifiers. What's the failure and how did you try to fix that?

aque commented:
Below is the output of the build. I took the first error (`implicit declaration of function 'OPENSSL_sk_find_all' is invalid in C99 [-Werror,-Wimplicit-function-declaration]`) and searched Google to see if there is any mention of similar issues.  Finding none, I looked at lines around 10631 within `_m2crypto_wrap.c` to see if there is anything that relates to Python versions. That is really all I have done as I am just trying to keep this port updated.

```
--->  Building py27-m2crypto
DEBUG: Tracelib Sandbox is:
DEBUG: 	/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports-ports_python_py-m2crypto/py27-m2crypto/work=+
DEBUG: 	/opt/local/var/macports/sources/github.com/macports-ports/python/py-m2crypto=+
DEBUG: 	/opt/local/var/macports/distfiles/py-m2crypto=+
DEBUG: 	/bin=+
DEBUG: 	/sbin=+
DEBUG: 	/dev=+
DEBUG: 	/usr/bin=+
DEBUG: 	/usr/sbin=+
DEBUG: 	/usr/include=+
DEBUG: 	/usr/lib=+
DEBUG: 	/usr/libexec=+
DEBUG: 	/usr/share=+
DEBUG: 	/System/Library=+
DEBUG: 	/Library/Frameworks=-
DEBUG: 	/Library=+
DEBUG: 	/etc=+
DEBUG: 	/private/etc=+
DEBUG: 	/tmp=+
DEBUG: 	/private/tmp=+
DEBUG: 	/var/tmp=+
DEBUG: 	/private/var/tmp=+
DEBUG: 	/var/folders=+
DEBUG: 	/private/var/folders=+
DEBUG: 	/var/empty=+
DEBUG: 	/private/var/empty=+
DEBUG: 	/var/run=+
DEBUG: 	/private/var/run=+
DEBUG: 	/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports-ports_python_py-m2crypto/py27-m2crypto/work/.tmp=+
DEBUG: 	/var/db/timezone/zoneinfo=+
DEBUG: 	/private/var/db/timezone/tz/2023c.1.0/zoneinfo=+
DEBUG: 	/var/db/mds/system=+
DEBUG: 	/private/var/db/mds/system=+
DEBUG: 	/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk=+
DEBUG: 	/Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk=+
DEBUG: 	/var/db/xcode_select_link=-
DEBUG: 	/private/var/db/xcode_select_link=-
DEBUG: 	/var/db/mds=-
DEBUG: 	/private/var/db/mds=-
DEBUG: 	/opt/local/var/macports/home/Library/Preferences/com.apple.dt.Xcode.plist=-
DEBUG: 	/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports-ports_python_py-m2crypto/py27-m2crypto/work/.home/Library/Preferences/com.apple.dt.Xcode.plist=-
DEBUG: 	/Applications/Xcode.app=-
DEBUG: 	/var/db/launchd.db=+
DEBUG: 	/private/var/db/launchd.db=+
DEBUG: 	/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports-ports_python_py-m2crypto/py27-m2crypto/work/.home/.ccache=+
DEBUG: 	/opt/local/var/macports/build/.ccache=+
DEBUG: 	/opt/local/var/macports/sip-workaround=+
DEBUG: 	/opt/local=?
DEBUG: Tracemode will respect recursively collected port dependencies: bzip2 db48 expat gettext-runtime libedit libffi libiconv ncurses openssl openssl11 openssl3 pcre2 py27-setuptools py27-typing python27 python2_select python_select sqlite3 swig swig-python zlib
DEBUG: Executing proc-pre-org.macports.build-build-0
DEBUG: Executing org.macports.build (py27-m2crypto)
DEBUG: Environment: 
CC='/usr/bin/clang'
CC_PRINT_OPTIONS='YES'
CC_PRINT_OPTIONS_FILE='/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports-ports_python_py-m2crypto/py27-m2crypto/work/.CC_PRINT_OPTIONS'
CFLAGS='-arch x86_64 -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk'
CPATH='/opt/local/include'
CXX='/usr/bin/clang++'
CXXFLAGS='-arch x86_64 -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk'
DEVELOPER_DIR='/Library/Developer/CommandLineTools'
F90FLAGS='-m64'
FCFLAGS='-m64'
FFLAGS='-m64'
LDFLAGS='-arch x86_64'
LIBRARY_PATH='/opt/local/lib'
MACOSX_DEPLOYMENT_TARGET='11.0'
OBJC='/usr/bin/clang'
OBJCFLAGS='-arch x86_64 -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk'
SDKROOT='/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk'
SWIG_FEATURES='-I/opt/local/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include -cpperraswarn'
Executing:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports-ports_python_py-m2crypto/py27-m2crypto/work/M2Crypto-0.39.0" && /opt/local/Library/Frameworks/Python.framework/Versions/2.7/bin/python2.7 setup.py --no-user-cfg build_ext --openssl=/opt/local build 
DEBUG: system:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports-ports_python_py-m2crypto/py27-m2crypto/work/M2Crypto-0.39.0" && /opt/local/Library/Frameworks/Python.framework/Versions/2.7/bin/python2.7 setup.py --no-user-cfg build_ext --openssl=/opt/local build 
running build_ext
creating build
creating build/lib.macosx-11.0-x86_64-2.7
creating build/lib.macosx-11.0-x86_64-2.7/M2Crypto
building 'M2Crypto._m2crypto' extension
swigging src/SWIG/_m2crypto.i to src/SWIG/_m2crypto_wrap.c
swig -python -I/opt/local/include -I/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include -I/Library/Developer/CommandLineTools/usr/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/System/Library/Frameworks (framework directory) -I/opt/local/include -I/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include -I/Library/Developer/CommandLineTools/usr/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/System/Library/Frameworks (framework directory) -I/opt/local/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -I/opt/local/include -I/opt/local/include/openssl -includeall -modern -builtin -outdir /opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports-ports_python_py-m2crypto/py27-m2crypto/work/M2Crypto-0.39.0/src/M2Crypto -o src/SWIG/_m2crypto_wrap.c src/SWIG/_m2crypto.i
Deprecated command line option: -modern. Ignored, this option is now always on.
/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/sys/cdefs.h:81: Warning 204: CPP #warning, ""Unsupported compiler detected"".
/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/sys/cdefs.h:828: Warning 205: CPP #error "Unsupported architecture".
/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/machine/_types.h:36: Warning 205: CPP #error "architecture not supported".
/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/machine/types.h:39: Warning 205: CPP #error "architecture not supported".
creating build/temp.macosx-11.0-x86_64-2.7
creating build/temp.macosx-11.0-x86_64-2.7/src
creating build/temp.macosx-11.0-x86_64-2.7/src/SWIG
/usr/bin/clang -fno-strict-aliasing -fno-common -dynamic -pipe -I/opt/local/libexec/openssl11/include -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -arch x86_64 -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk -I/opt/local/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -I/opt/local/include -I/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports-ports_python_py-m2crypto/py27-m2crypto/work/M2Crypto-0.39.0/src/SWIG -c src/SWIG/_m2crypto_wrap.c -o build/temp.macosx-11.0-x86_64-2.7/src/SWIG/_m2crypto_wrap.o -Wno-deprecated-declarations -DTHREADING
src/SWIG/_m2crypto_wrap.c:4859:9: warning: 'PyUnicode_FromString' macro redefined [-Wmacro-redefined]
#define PyUnicode_FromString(x) PyString_FromString(x)
        ^
/opt/local/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7/unicodeobject.h:194:10: note: previous definition is here
# define PyUnicode_FromString PyUnicodeUCS2_FromString
         ^
src/SWIG/_m2crypto_wrap.c:4860:9: warning: 'PyUnicode_Format' macro redefined [-Wmacro-redefined]
#define PyUnicode_Format(x, y) PyString_Format(x, y)
        ^
/opt/local/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7/unicodeobject.h:188:10: note: previous definition is here
# define PyUnicode_Format PyUnicodeUCS2_Format
         ^
src/SWIG/_m2crypto_wrap.c:10631:17: error: implicit declaration of function 'OPENSSL_sk_find_all' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (int)OPENSSL_sk_find_all(arg1,(void const *)arg2,arg3);
                ^
src/SWIG/_m2crypto_wrap.c:10631:17: note: did you mean 'OPENSSL_sk_find_ex'?
/opt/local/libexec/openssl11/include/openssl/stack.h:41:5: note: 'OPENSSL_sk_find_ex' declared here
int OPENSSL_sk_find_ex(OPENSSL_STACK *st, const void *data);
    ^
src/SWIG/_m2crypto_wrap.c:12492:20: error: implicit declaration of function 'ossl_check_OPENSSL_STRING_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (char *)ossl_check_OPENSSL_STRING_type(arg1);
                   ^
src/SWIG/_m2crypto_wrap.c:12492:20: note: did you mean '_wrap_ossl_check_OPENSSL_STRING_type'?
src/SWIG/_m2crypto_wrap.c:12475:22: note: '_wrap_ossl_check_OPENSSL_STRING_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_OPENSSL_STRING_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12492:12: warning: cast to 'char *' from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (char *)ossl_check_OPENSSL_STRING_type(arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12518:29: error: implicit declaration of function 'ossl_check_const_OPENSSL_STRING_sk_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (OPENSSL_STACK *)ossl_check_const_OPENSSL_STRING_sk_type((struct stack_st_OPENSSL_STRING const *)arg1);
                            ^
src/SWIG/_m2crypto_wrap.c:12518:29: note: did you mean '_wrap_ossl_check_const_OPENSSL_STRING_sk_type'?
src/SWIG/_m2crypto_wrap.c:12502:22: note: '_wrap_ossl_check_const_OPENSSL_STRING_sk_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_const_OPENSSL_STRING_sk_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12518:12: warning: cast to 'OPENSSL_STACK *' (aka 'struct stack_st *') from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (OPENSSL_STACK *)ossl_check_const_OPENSSL_STRING_sk_type((struct stack_st_OPENSSL_STRING const *)arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12542:29: error: implicit declaration of function 'ossl_check_OPENSSL_STRING_sk_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (OPENSSL_STACK *)ossl_check_OPENSSL_STRING_sk_type(arg1);
                            ^
src/SWIG/_m2crypto_wrap.c:12542:29: note: did you mean '_wrap_ossl_check_OPENSSL_STRING_sk_type'?
src/SWIG/_m2crypto_wrap.c:12526:22: note: '_wrap_ossl_check_OPENSSL_STRING_sk_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_OPENSSL_STRING_sk_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12542:12: warning: cast to 'OPENSSL_STACK *' (aka 'struct stack_st *') from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (OPENSSL_STACK *)ossl_check_OPENSSL_STRING_sk_type(arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12565:33: error: implicit declaration of function 'ossl_check_OPENSSL_STRING_compfunc_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (OPENSSL_sk_compfunc)ossl_check_OPENSSL_STRING_compfunc_type(arg1);
                                ^
src/SWIG/_m2crypto_wrap.c:12565:33: note: did you mean '_wrap_ossl_check_OPENSSL_STRING_compfunc_type'?
src/SWIG/_m2crypto_wrap.c:12550:22: note: '_wrap_ossl_check_OPENSSL_STRING_compfunc_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_OPENSSL_STRING_compfunc_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12565:12: warning: cast to 'OPENSSL_sk_compfunc' (aka 'int (*)(const void *, const void *)') from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (OPENSSL_sk_compfunc)ossl_check_OPENSSL_STRING_compfunc_type(arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12588:33: error: implicit declaration of function 'ossl_check_OPENSSL_STRING_copyfunc_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (OPENSSL_sk_copyfunc)ossl_check_OPENSSL_STRING_copyfunc_type(arg1);
                                ^
src/SWIG/_m2crypto_wrap.c:12588:33: note: did you mean '_wrap_ossl_check_OPENSSL_STRING_copyfunc_type'?
src/SWIG/_m2crypto_wrap.c:12573:22: note: '_wrap_ossl_check_OPENSSL_STRING_copyfunc_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_OPENSSL_STRING_copyfunc_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12588:12: warning: cast to 'OPENSSL_sk_copyfunc' (aka 'void *(*)(const void *)') from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (OPENSSL_sk_copyfunc)ossl_check_OPENSSL_STRING_copyfunc_type(arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12611:33: error: implicit declaration of function 'ossl_check_OPENSSL_STRING_freefunc_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (OPENSSL_sk_freefunc)ossl_check_OPENSSL_STRING_freefunc_type(arg1);
                                ^
src/SWIG/_m2crypto_wrap.c:12611:33: note: did you mean '_wrap_ossl_check_OPENSSL_STRING_freefunc_type'?
src/SWIG/_m2crypto_wrap.c:12596:22: note: '_wrap_ossl_check_OPENSSL_STRING_freefunc_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_OPENSSL_STRING_freefunc_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12611:12: warning: cast to 'OPENSSL_sk_freefunc' (aka 'void (*)(void *)') from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (OPENSSL_sk_freefunc)ossl_check_OPENSSL_STRING_freefunc_type(arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12636:20: error: implicit declaration of function 'ossl_check_OPENSSL_CSTRING_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (char *)ossl_check_OPENSSL_CSTRING_type((char const *)arg1);
                   ^
src/SWIG/_m2crypto_wrap.c:12636:20: note: did you mean '_wrap_ossl_check_OPENSSL_CSTRING_type'?
src/SWIG/_m2crypto_wrap.c:12619:22: note: '_wrap_ossl_check_OPENSSL_CSTRING_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_OPENSSL_CSTRING_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12636:12: warning: cast to 'char *' from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (char *)ossl_check_OPENSSL_CSTRING_type((char const *)arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12662:29: error: implicit declaration of function 'ossl_check_const_OPENSSL_CSTRING_sk_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (OPENSSL_STACK *)ossl_check_const_OPENSSL_CSTRING_sk_type((struct stack_st_OPENSSL_CSTRING const *)arg1);
                            ^
src/SWIG/_m2crypto_wrap.c:12662:29: note: did you mean '_wrap_ossl_check_const_OPENSSL_CSTRING_sk_type'?
src/SWIG/_m2crypto_wrap.c:12646:22: note: '_wrap_ossl_check_const_OPENSSL_CSTRING_sk_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_const_OPENSSL_CSTRING_sk_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12662:12: warning: cast to 'OPENSSL_STACK *' (aka 'struct stack_st *') from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (OPENSSL_STACK *)ossl_check_const_OPENSSL_CSTRING_sk_type((struct stack_st_OPENSSL_CSTRING const *)arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12686:29: error: implicit declaration of function 'ossl_check_OPENSSL_CSTRING_sk_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (OPENSSL_STACK *)ossl_check_OPENSSL_CSTRING_sk_type(arg1);
                            ^
src/SWIG/_m2crypto_wrap.c:12686:29: note: did you mean '_wrap_ossl_check_OPENSSL_CSTRING_sk_type'?
src/SWIG/_m2crypto_wrap.c:12670:22: note: '_wrap_ossl_check_OPENSSL_CSTRING_sk_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_OPENSSL_CSTRING_sk_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12686:12: warning: cast to 'OPENSSL_STACK *' (aka 'struct stack_st *') from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (OPENSSL_STACK *)ossl_check_OPENSSL_CSTRING_sk_type(arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12709:33: error: implicit declaration of function 'ossl_check_OPENSSL_CSTRING_compfunc_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (OPENSSL_sk_compfunc)ossl_check_OPENSSL_CSTRING_compfunc_type(arg1);
                                ^
src/SWIG/_m2crypto_wrap.c:12709:33: note: did you mean '_wrap_ossl_check_OPENSSL_CSTRING_compfunc_type'?
src/SWIG/_m2crypto_wrap.c:12694:22: note: '_wrap_ossl_check_OPENSSL_CSTRING_compfunc_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_OPENSSL_CSTRING_compfunc_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12709:12: warning: cast to 'OPENSSL_sk_compfunc' (aka 'int (*)(const void *, const void *)') from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (OPENSSL_sk_compfunc)ossl_check_OPENSSL_CSTRING_compfunc_type(arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12732:33: error: implicit declaration of function 'ossl_check_OPENSSL_CSTRING_copyfunc_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (OPENSSL_sk_copyfunc)ossl_check_OPENSSL_CSTRING_copyfunc_type(arg1);
                                ^
src/SWIG/_m2crypto_wrap.c:12732:12: warning: cast to 'OPENSSL_sk_copyfunc' (aka 'void *(*)(const void *)') from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (OPENSSL_sk_copyfunc)ossl_check_OPENSSL_CSTRING_copyfunc_type(arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12755:33: error: implicit declaration of function 'ossl_check_OPENSSL_CSTRING_freefunc_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (OPENSSL_sk_freefunc)ossl_check_OPENSSL_CSTRING_freefunc_type(arg1);
                                ^
src/SWIG/_m2crypto_wrap.c:12755:33: note: did you mean '_wrap_ossl_check_OPENSSL_CSTRING_freefunc_type'?
src/SWIG/_m2crypto_wrap.c:12740:22: note: '_wrap_ossl_check_OPENSSL_CSTRING_freefunc_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_OPENSSL_CSTRING_freefunc_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12755:12: warning: cast to 'OPENSSL_sk_freefunc' (aka 'void (*)(void *)') from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (OPENSSL_sk_freefunc)ossl_check_OPENSSL_CSTRING_freefunc_type(arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12777:20: error: implicit declaration of function 'ossl_check_OPENSSL_BLOCK_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (void *)ossl_check_OPENSSL_BLOCK_type(arg1);
                   ^
src/SWIG/_m2crypto_wrap.c:12777:20: note: did you mean '_wrap_ossl_check_OPENSSL_BLOCK_type'?
src/SWIG/_m2crypto_wrap.c:12763:22: note: '_wrap_ossl_check_OPENSSL_BLOCK_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_OPENSSL_BLOCK_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12777:12: warning: cast to 'void *' from smaller integer type 'int' [-Wint-to-void-pointer-cast]
  result = (void *)ossl_check_OPENSSL_BLOCK_type(arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12801:29: error: implicit declaration of function 'ossl_check_const_OPENSSL_BLOCK_sk_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (OPENSSL_STACK *)ossl_check_const_OPENSSL_BLOCK_sk_type((struct stack_st_OPENSSL_BLOCK const *)arg1);
                            ^
src/SWIG/_m2crypto_wrap.c:12801:29: note: did you mean '_wrap_ossl_check_const_OPENSSL_BLOCK_sk_type'?
src/SWIG/_m2crypto_wrap.c:12785:22: note: '_wrap_ossl_check_const_OPENSSL_BLOCK_sk_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_const_OPENSSL_BLOCK_sk_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12801:12: warning: cast to 'OPENSSL_STACK *' (aka 'struct stack_st *') from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (OPENSSL_STACK *)ossl_check_const_OPENSSL_BLOCK_sk_type((struct stack_st_OPENSSL_BLOCK const *)arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12825:29: error: implicit declaration of function 'ossl_check_OPENSSL_BLOCK_sk_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (OPENSSL_STACK *)ossl_check_OPENSSL_BLOCK_sk_type(arg1);
                            ^
src/SWIG/_m2crypto_wrap.c:12825:29: note: did you mean '_wrap_ossl_check_OPENSSL_BLOCK_sk_type'?
src/SWIG/_m2crypto_wrap.c:12809:22: note: '_wrap_ossl_check_OPENSSL_BLOCK_sk_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_OPENSSL_BLOCK_sk_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12825:12: warning: cast to 'OPENSSL_STACK *' (aka 'struct stack_st *') from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (OPENSSL_STACK *)ossl_check_OPENSSL_BLOCK_sk_type(arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12848:33: error: implicit declaration of function 'ossl_check_OPENSSL_BLOCK_compfunc_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (OPENSSL_sk_compfunc)ossl_check_OPENSSL_BLOCK_compfunc_type(arg1);
                                ^
src/SWIG/_m2crypto_wrap.c:12848:33: note: did you mean '_wrap_ossl_check_OPENSSL_BLOCK_compfunc_type'?
src/SWIG/_m2crypto_wrap.c:12833:22: note: '_wrap_ossl_check_OPENSSL_BLOCK_compfunc_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_OPENSSL_BLOCK_compfunc_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12848:12: warning: cast to 'OPENSSL_sk_compfunc' (aka 'int (*)(const void *, const void *)') from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (OPENSSL_sk_compfunc)ossl_check_OPENSSL_BLOCK_compfunc_type(arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12871:33: error: implicit declaration of function 'ossl_check_OPENSSL_BLOCK_copyfunc_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (OPENSSL_sk_copyfunc)ossl_check_OPENSSL_BLOCK_copyfunc_type(arg1);
                                ^
src/SWIG/_m2crypto_wrap.c:12871:33: note: did you mean '_wrap_ossl_check_OPENSSL_BLOCK_copyfunc_type'?
src/SWIG/_m2crypto_wrap.c:12856:22: note: '_wrap_ossl_check_OPENSSL_BLOCK_copyfunc_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_OPENSSL_BLOCK_copyfunc_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12871:12: warning: cast to 'OPENSSL_sk_copyfunc' (aka 'void *(*)(const void *)') from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (OPENSSL_sk_copyfunc)ossl_check_OPENSSL_BLOCK_copyfunc_type(arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SWIG/_m2crypto_wrap.c:12894:33: error: implicit declaration of function 'ossl_check_OPENSSL_BLOCK_freefunc_type' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  result = (OPENSSL_sk_freefunc)ossl_check_OPENSSL_BLOCK_freefunc_type(arg1);
                                ^
src/SWIG/_m2crypto_wrap.c:12894:33: note: did you mean '_wrap_ossl_check_OPENSSL_BLOCK_freefunc_type'?
src/SWIG/_m2crypto_wrap.c:12879:22: note: '_wrap_ossl_check_OPENSSL_BLOCK_freefunc_type' declared here
SWIGINTERN PyObject *_wrap_ossl_check_OPENSSL_BLOCK_freefunc_type(PyObject *self, PyObject *args) {
                     ^
src/SWIG/_m2crypto_wrap.c:12894:12: warning: cast to 'OPENSSL_sk_freefunc' (aka 'void (*)(void *)') from smaller integer type 'int' [-Wint-to-pointer-cast]
  result = (OPENSSL_sk_freefunc)ossl_check_OPENSSL_BLOCK_freefunc_type(arg1);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
20 warnings and 19 errors generated.
error: command '/usr/bin/clang' failed with exit status 1
Command failed:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports-ports_python_py-m2crypto/py27-m2crypto/work/M2Crypto-0.39.0" && /opt/local/Library/Frameworks/Python.framework/Versions/2.7/bin/python2.7 setup.py --no-user-cfg build_ext --openssl=/opt/local build 
Exit code: 1
```

reneeotten commented:
okay, so the CI show that there is an "implicit function declaration"; that should be fixed, perhaps just missing an "include"'of a header. You'll need to look into to build/log a bit more to figure out where/what to do.

aque commented:
This is going over my head, but if it is missing an "include", the same error would happen in other versions and not limited to py27 and py37. The "implicit function declaration" is implicit in those working versions as well. I have just tried all previous versions going back to 0.32.0 and they all fail under py27 with the same error. I am thinking this issue is unique to Macports since Google yields no results.

aque commented:
So I dug into this some more. I sorted and compared the clang parameters between the non-working builds (py27 and py37) with two of the working ones (py38 and py310). The failed builds both have `-I/opt/local/libexec/openssl11/include ` while the working ones do not have it. This extra include might be the difference, although I do not know where that parameter is coming from.

py27 (not working)
```
/usr/bin/clang 
-DNDEBUG 
-DTHREADING
-I/opt/local/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 
-I/opt/local/include 
-I/opt/local/libexec/openssl11/include 
-I/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports-ports_python_py-m2crypto/py27-m2crypto/work/M2Crypto-0.39.0/src/SWIG 
-O3 
-Os 
-Wall 
-Wno-deprecated-declarations 
-Wstrict-prototypes 
-arch x86_64 
-c src/SWIG/_m2crypto_wrap.c 
-dynamic 
-fno-common 
-fno-strict-aliasing 
-fwrapv 
-g 
-isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk 
-isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk 
-o build/temp.macosx-11.0-x86_64-2.7/src/SWIG/_m2crypto_wrap.o 
-pipe 
```

py37 (not working)
```
/usr/bin/clang
-DNDEBUG 
-DTHREADING
-I/opt/local/Library/Frameworks/Python.framework/Versions/3.7/include/python3.7m 
-I/opt/local/include 
-I/opt/local/libexec/openssl11/include 
-I/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports-ports_python_py-m2crypto/py37-m2crypto/work/M2Crypto-0.39.0/src/SWIG 
-O3 
-Os 
-Wall 
-Wno-deprecated-declarations 
-Wno-unused-result 
-Wsign-compare 
-Wunreachable-code 
-arch x86_64 
-c src/SWIG/_m2crypto_wrap.c 
-dynamic 
-fno-common 
-fwrapv 
-g 
-isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk 
-isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk 
-o build/temp.macosx-11.0-x86_64-cpython-37/src/SWIG/_m2crypto_wrap.o 
-pipe 
```

py38 (working)
```
/usr/bin/clang 
-DNDEBUG 
-DTHREADING 
-I/opt/local/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8 
-I/opt/local/include 
-I/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports-ports_python_py-m2crypto/py38-m2crypto/work/M2Crypto-0.39.0/src/SWIG 
-O3 
-Os 
-Wall 
-Wno-deprecated-declarations
-Wno-unused-result 
-Wsign-compare 
-Wunreachable-code 
-arch x86_64 
-c src/SWIG/_m2crypto_wrap.c 
-dynamic 
-fno-common 
-fwrapv 
-g 
-isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk 
-isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk 
-o build/temp.macosx-11.0-x86_64-cpython-38/src/SWIG/_m2crypto_wrap.o 
-pipe 
```

py310 (working)
```
/usr/bin/clang 
-DNDEBUG 
-DTHREADING
-I/opt/local/Library/Frameworks/Python.framework/Versions/3.10/include/python3.10 
-I/opt/local/include 
-I/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports-ports_python_py-m2crypto/py310-m2crypto/work/M2Crypto-0.39.0/src/SWIG 
-O3 
-Os 
-Wall 
-Wno-deprecated-declarations 
-Wno-unused-result 
-Wsign-compare 
-Wunreachable-code 
-arch x86_64 
-c src/SWIG/_m2crypto_wrap.c 
-dynamic 
-fno-common 
-fwrapv 
-g 
-isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk 
-isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk 
-o build/temp.macosx-11.0-x86_64-cpython-310/src/SWIG/_m2crypto_wrap.o 
-pipe 
```

jmroot commented:
Only python 3.7 and older use openssl11 because they don't work with newer versions. It's perhaps questionable whether all the CPPFLAGS used for building python itself should be used for building modules as well, but the real issue here may be the include order:
```
-I/opt/local/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 
-I/opt/local/include 
-I/opt/local/libexec/openssl11/include 
-I/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports-ports_python_py-m2crypto/py27-m2crypto/work/M2Crypto-0.39.0/src/SWIG 
```
Include directories specified with `-I` are searched in the order they are given on the command line. Internal directories like the last one there should almost always come first, as otherwise an unrelated header with the same name could be used instead of the intended internal one.

aque commented:
@jmroot I want to clarify that the include directories above were sorted to look for differences between runs. The actual order from the py27 build is below. Just making sure this info is accurate since you said the order is important.

```
-I/opt/local/libexec/openssl11/include
-I/opt/local/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 
-I/opt/local/include 
-I/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports-ports_python_py-m2crypto/py27-m2crypto/work/M2Crypto-0.39.0/src/SWIG 
```

reneeotten commented:
Based on this and Josh his response that order is likely wrong and the last one listed here should actually be the first one. Try to change the order of the includes in the build script and/or Portfile and try again to see if that fixes the build.

jmroot commented:
The setup.py has some reasonably complicated logic for setting up includes, but it may be as simple as patching this line to change it from `.append(blah)` to (.insert(0, blah)`: https://gitlab.com/m2crypto/m2crypto/-/blob/master/setup.py?ref_type=heads#L223

aque commented:
The last accepted version (0.38.0 from #12531) has the same include order. Builds would fail then if that was the issue. The port builds correctly for python3.5 and python3.6, and those versions do not list `-I /opt/local/libexec/openssl11/include` as one of their clang includes.

I spent a lot of time finding a way to filter that extra include but had no luck. Then, Josh’s comment provided the breakthrough that it was backwards - *python3.5 and 3.6 both need an openss11 include and was missing*; python2.7 and 3.7 were the correct way. I also did not know older pythons required openssl11, and finding the openssl portgroup helped tremendously.

I updated the commit and it now passed the CI. I also cleaned up the port and fixed the whitespaces. Hopefully, we can now merge this PR.
